pandas/_libs/lib.pyx

```diff
@@ -1210,6 +1210,28 @@ try:
 except AttributeError:
     pass
 
+cdef enum CurrSeen:
```

Comment:
Can you combine `CurrSeen` and `CurrSeenVal` into one struct? I think it would help readability to manage all within one object instead of separate

Reply:
I just pushed something, so that NA value handling goes through the ``Seen`` object as well. 

The logic should be greatly simplified now since everything goes through ``Seen`` now.

Reply:
Very nice - seems like a much better home for this